src/diffusion_for_multi_scale_molecular_dynamics/models/axl_diffusion_lightning_model.py

```diff
@@ -180,73 +254,164 @@ def _generic_step(
+            f"the shape of the ATOM_TYPES array should be [batch_size, number_of_atoms]. "
+            f"Got shape = {atom_shape}"
+        )
+
+        lvec0 = batch[
+            "box"
+        ]  # should be batch[UNIT_CELL] - see later comment with batch['box']
+        # TODO assert on shape
 
-        noise_sample = self.variance_sampler.get_random_noise_sample(batch_size)
+        noise_sample = self.noise_scheduler.get_random_noise_sample(batch_size)
 
-        # noise_sample.sigma has  dimension [batch_size]. Broadcast these sigma values to be
-        # of shape [batch_size, number_of_atoms, spatial_dimension], which can be interpreted
-        # as [batch_size, (configuration)]. All the sigma values must be the same for a given configuration.
+        # noise_sample.sigma and has dimension [batch_size]. Broadcast these values to be of shape
+        # [batch_size, number_of_atoms, spatial_dimension] , which can be interpreted as
+        # [batch_size, (configuration)]. All the sigma values must be the same for a given configuration.
         sigmas = broadcast_batch_tensor_to_all_dimensions(
             batch_values=noise_sample.sigma, final_shape=shape
         )
+        # we can now get noisy coordinates
+        xt = self.noisers.X.get_noisy_relative_coordinates_sample(x0, sigmas)
+
+        # to get noisy atom types, we need to broadcast the transition matrix q_bar from size
+        # [num_atom_types, num_atom_types] to [batch_size, number_of_atoms, num_atom_types, num_atom_types]. All the
+        # q_bar matrices must be the same for a given configuration.
+        q_matrices = broadcast_batch_matrix_tensor_to_all_dimensions(
+            batch_values=noise_sample.q_matrix, final_shape=atom_shape
+        )
+        q_bar_matrices = broadcast_batch_matrix_tensor_to_all_dimensions(
+            batch_values=noise_sample.q_bar_matrix, final_shape=atom_shape
+        )
 
-        xt = self.noisy_relative_coordinates_sampler.get_noisy_relative_coordinates_sample(
-            x0, sigmas
+        q_bar_tm1_matrices = broadcast_batch_matrix_tensor_to_all_dimensions(
+            batch_values=noise_sample.q_bar_tm1_matrix, final_shape=atom_shape
         )
 
-        # The target is nabla log p_{t|0} (xt | x0): it is NOT the "score", but rather a "conditional" (on x0) score.
-        target_normalized_conditional_scores = self._get_target_normalized_score(
-            xt, x0, sigmas
+        # we also need the atom types to be one-hot vector and not a class index
+        a0_onehot = class_index_to_onehot(a0, self.num_atom_types + 1)
+
+        at = self.noisers.A.get_noisy_atom_types_sample(a0_onehot, q_bar_matrices)
+        at_onehot = class_index_to_onehot(at, self.num_atom_types + 1)
+
+        # TODO do the same for the lattice vectors
+        lvect = self.noisers.L.get_noisy_lattice_vectors(lvec0)
+
+        noisy_sample = AXL(A=at, X=xt, L=lvec0)  # not one-hot
+
+        original_sample = AXL(A=a0, X=x0, L=lvect)
```

Comment:
I propose calling this `composition` instead of `original_sample`. See my comments in `namespace.py`.